src/test/java/pro/trevor/tankgame/Rule/BuyActionWithGoldTest.java

```diff
@@ -0,0 +1,166 @@
+package pro.trevor.tankgame.Rule;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import pro.trevor.tankgame.rule.definition.player.PlayerActionRule;
+import pro.trevor.tankgame.rule.impl.shared.PlayerRules;
+import pro.trevor.tankgame.rule.impl.version3.Tank;
+import pro.trevor.tankgame.rule.impl.version3.TankAttribute;
+import pro.trevor.tankgame.state.State;
+
+public class BuyActionWithGoldTest 
+{
+    @Test
+    public void TypeCheckTest()
+    {
+        PlayerActionRule<Tank> rule = PlayerRules.BuyActionWithGold(3); // action_cost is a don't care
+        assertEquals(PlayerRules.ActionKeys.BUY_ACTION, rule.name(), "Asserts that the buy action with gold rule is the right type");
+    }
+
+    private Tank BuildTestTank(int actions, int gold, boolean dead)
+    {
+        JSONObject json = new JSONObject();
+        JSONObject attributes = new JSONObject();
+        attributes.put(TankAttribute.ACTIONS.name(), actions);
+        attributes.put(TankAttribute.GOLD.name(), gold);
+        attributes.put(TankAttribute.DEAD.name(), dead);
+        json.put("name", "test");
+        json.put("attributes", attributes);
+        json.put("type", "tank");
+        json.put("position", "A1");
+        return new Tank(json);
+    }
+
+    @Test
+    public void DeadTankCannotBuyAction()
+    {
+        Tank tank = BuildTestTank(0, 3, true);
+
+        // Get 3 gold-cost rule
```

Comment:
These comments seem redundant for all of these tests. Maybe comments with the expected state of the tanks before/after applications are fine.